code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,345 @@
+	if(!(length(create_spiderling_action.spiderlings)))
+		X.balloon_alert(X, "No spiderlings")
+		return fail_activate()
+	var/list/mob/living/carbon/xenomorph/spiderling/remaining_spiderlings = create_spiderling_action.spiderlings.Copy()
+	grab_spiderlings(remaining_spiderlings, attach_attempts)
+	succeed_activate()
+
+/// this proc scoops up adjacent spiderlings and then calls ride_widow on them
+/datum/action/xeno_action/attach_spiderlings/proc/grab_spiderlings(list/mob/living/carbon/xenomorph/spiderling/remaining_list, number_of_attempts_left)
+	if(number_of_attempts_left <= 0)
+		return
+	for(var/mob/living/carbon/xenomorph/spiderling/remaining_spiderling AS in remaining_list)
+		if(!owner.Adjacent(remaining_spiderling))
+			continue
+		remaining_list -= remaining_spiderling
+		owner.buckle_mob(remaining_spiderling, TRUE, TRUE, 90, 1,0)
+	addtimer(CALLBACK(src, .proc/grab_spiderlings, remaining_list, number_of_attempts_left - 1), 1)
+
+// ***************************************
+// *********** Web Hook
+// ***************************************
+/datum/action/xeno_action/activable/web_hook
+	name = "Web Hook"
+	ability_name = "Web Hook"
+	mechanics_text = "Shoot out a web and pull it to traverse forward"
+	action_icon_state = "web_hook"
+	plasma_cost = 200
+	cooldown_timer = 20 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_WEB_HOOK
+	//ref to beam for web hook
+	var/datum/beam/web_beam
+
+/datum/action/xeno_action/activable/web_hook/can_use_ability(atom/A)
+	. = ..()
+	if(!.)
+		return
+	if(isliving(A))
+		owner.balloon_alert(owner, "We can't attach to that")
+		return FALSE
+	if(!isturf(A))
+		return FALSE
+	if(get_dist(owner, A) <= WIDOW_WEB_HOOK_MIN_RANGE)
+		owner.balloon_alert(owner, "Too close")
+		return FALSE
+	var/turf/current = get_turf(owner)
+	var/turf/target_turf = get_turf(A)
+	if(get_dist(current, target_turf) > WIDOW_WEB_HOOK_RANGE)
+		owner.balloon_alert(owner, "Too far")
+		return FALSE
+	current = get_step_towards(current, target_turf)
+	while((current != target_turf))
+		current = get_step_towards(current, target_turf)
+
+/datum/action/xeno_action/activable/web_hook/use_ability(atom/A)
+	var/atom/movable/web_hook/web_hook = new (get_turf(owner))
+	web_beam = owner.beam(web_hook,"beam_web",'icons/effects/beam.dmi')
+	RegisterSignal(web_hook, list(COMSIG_MOVABLE_POST_THROW, COMSIG_MOVABLE_IMPACT), .proc/drag_widow, TRUE)
+	web_hook.throw_at(A, WIDOW_WEB_HOOK_RANGE, 3, owner, FALSE)
+	succeed_activate()
+	add_cooldown()
+
+/// This throws widow wherever the web_hook landed, distance is dependant on if the web_hook hit a wall or just ground
+/datum/action/xeno_action/activable/web_hook/proc/drag_widow(datum/source, turf/t)
+	SIGNAL_HANDLER
+	QDEL_NULL(web_beam)
+	if(t)
+		owner.throw_at(t, WIDOW_WEB_HOOK_RANGE, WIDOW_WEB_HOOK_SPEED, owner, FALSE)
+	else
+		/// we throw widow half the distance if she hits the floor
+		owner.throw_at(get_turf(source), WIDOW_WEB_HOOK_RANGE / 2, WIDOW_WEB_HOOK_SPEED, owner, FALSE)
```

Comment:
bad var names use a proper name
also done use autodoc format for non autodoc comments it looks wierd